### PR TITLE
refactor: to make exchange adapter

### PR DESCRIPTION
**Pull request by ejfitzgerald**
## Summary

Some code refactoring to make it easier to build standalone scripts to send exchange messages

### CODE REVIEW

File: python/src/uagents/context.py [hunk 1]
```diff
@@ -349,7 +349,8 @@ async def experimental_broadcast(
         agents = self.get_agents_by_protocol(destination_protocol, limit=limit)
         if not agents:
             self.logger.error(f"No active agents found for: {destination_protocol}")
-            return
+            return []
```
Comment (ejfitzgerald) [inline python/src/uagents/context.py:352]: My pycharm type checker was not a fan of this @Archento please check that I am not breaking something

File: python/src/uagents/context.py [hunk 2]
```diff
@@ -364,7 +365,7 @@ async def experimental_broadcast(
             ]
         )
         self.logger.debug(f"Sent {len(futures)} messages")
-        return futures
+        return list(futures)
```
Comment (ejfitzgerald) [inline python/src/uagents/context.py:368]: Likewise here @Archento I think that `asyncio.gather` returns a `tuple()` - can you and @jrriehl give this a double check

File: python/src/uagents/agent.py [hunk 3]
```diff
@@ -182,7 +183,11 @@ def __init__(
             if resolve is not None
             else GlobalResolver(max_endpoints=max_resolver_endpoints)
         )
-        self._loop = asyncio.get_event_loop_policy().get_event_loop()
+
+        if loop is not None:
+            self._loop = loop
```
Comment (qati) [inline python/src/uagents/agent.py:188]: Nice!

File: python/src/uagents/context.py [hunk 4]
```diff
@@ -457,12 +458,38 @@ async def send_raw(
...
+            schema_digest,
+            self.get_message_protocol(schema_digest),
+            json_message,
+            logger=self._logger,
+            timeout=timeout,
+            session_id=self._session,
+        )
+
+    @staticmethod
+    async def send_raw_exchange_envelope(
```
Comment (qati) [inline python/src/uagents/context.py:474]: Nice! If this could be released as 0.8.1 would be nice

**Review by jrriehl — APPROVED** [review]
Yeah, makes sense to me. I partly wonder if this should have been the original `send_raw` function, but I guess it's better to leave that in as well so as not to break anything.

File: python/src/uagents/context.py (hunk 2 above)
Comment (Archento) [reply python/src/uagents/context.py:368]: I just checked the example with 0..2 responses and all were lists.
According to the [docs](https://docs.python.org/3/library/asyncio-task.html#running-tasks-concurrently) all successful awaitables will be aggregated to a list and as we always return with a `MsgStatus` this should therefore also always be a list meaning that we shouldn't need the explicit casting although it doesn't hurt either.

**Review by Archento — DISMISSED** [review]

File: python/src/uagents/context.py (hunk 2 above)
Comment (jrriehl) [reply python/src/uagents/context.py:368]: Reverting this change due to @Archento's findings.

**Review by jrriehl — APPROVED** [review]

**Pull request merged**